src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -2133,12 +2149,28 @@ internal static bool SkipSystemRestoreOperationForARMPlatform(PSCmdlet cmdlet)
 
                     if (!InternalTestHooks.TestStopComputer)
                     {
-                        CimMethodResult result = cimSession.InvokeMethod(
-                            ComputerWMIHelper.CimOperatingSystemNamespace,
-                            ComputerWMIHelper.WMI_Class_OperatingSystem,
-                            ComputerWMIHelper.CimOperatingSystemShutdownMethod,
-                            methodParameters,
-                            operationOptions);
+                        CimMethodResult result = null;
+
+                        if (isLocalhost)
+                        {
+                            var computerSystem = cimSession.EnumerateInstances(ComputerWMIHelper.CimOperatingSystemNamespace, ComputerWMIHelper.WMI_Class_OperatingSystem);
```

Comment:
Add comment that Win32_OperatingSystem is a singleton

Reply:
Added.

Reply:
I'd prefer:
```suggestion
                            // Win32_ComputerSystem is a singleton hence FirstOrDefault() return the only instance returned by EnumerateInstances.
                            var computerSystem = cimSession.EnumerateInstances(ComputerWMIHelper.CimOperatingSystemNamespace, ComputerWMIHelper.WMI_Class_OperatingSystem).FirstOrDefault();
```